Control nodes multiply in size when child animationPlayer contains RESET track and a different scene is launched

### Tested versions

- Reproducible in Godot Engine v4.2.stable.mono.official [46dc27791]

_Bugsquad edited:_
- Reproducible in Godot Engine v4.2.stable
- Reproducible in Godot Engine v4.1.stable
- Reproducible in Godot Engine v4.0.stable

### System information

Godot v4.2.stable.mono - Windows 10.0.19045 - Vulkan (Mobile) - dedicated NVIDIA GeForce RTX 3090 Ti (NVIDIA; 31.0.15.5186) - 12th Gen Intel(R) Core(TM) i9-12900K (24 Threads)

### Issue description

Certain conditions cause nodes to erroneously change size. When this happens, each time the steps are reproduced, the relevant nodes will increase in size by a multiple of the original size - i.e. the first time they will end up 2x the original, then 3x, then 4x and so on.

### Steps to reproduce

The hierarchy [looks like this](https://imgur.com/LVxtMZC) where you have containers that structure the UI, an empty control node as a child to the container, and the another kind of node as a child of the empty control node. This last child also has an AnimationPlayer with a RESET track. This is to allow the other kind of node to be animated. Finally, set the control node and its child to use the anchor layout mode with the anchor preset set to Full Rect. I have tested this with Buttons, Labels, and LineEdits.

All of the above needs to be in 1 scene. Then while that scene is open in one tab, open a different scene in another tab and use **Run Current Scene** so the different scene is launched. You can immediately close that and go back to the original tab. In my example, [this is what it would have started looking like](https://imgur.com/Lbd4Nxu) and [this is how it will end up](https://i.imgur.com/XSXgzso.png). The new size of that node will be exactly 2x the starting size. If you repeat the process and launch the different scene again and come back, the node will now be 3x the starting size.

I recorded a video of this occurring, but I don't know how long it will remain available: https://streamable.com/5elh96.

### Minimal reproduction project (MRP)

[NodeSizeBug.zip](https://github.com/godotengine/godot/files/14817234/NodeSizeBug.zip)

Simply open both the main and different scenes so there are 2 tabs open. Launch the different scene with F6, then kill it. Go back to the main tab and the button should be a multiple bigger.

What do I have to do to propagate a notification?
I've tryed to do something like NOTIFICATION_EDITOR_PRE_SAVE but in the method _reset_animation_mixers.
I've also defined it in Node.xml, node.h and bind it in node.cpp, however Im gettign the error:
------------------------------------------------------------------------------
editor/editor_node.cpp:1770:36: error: 'NOTIFICATION_EDITOR_PRE_RESET' was not declared in this scope
 1770 |     p_node->propagate_notification(NOTIFICATION_EDITOR_PRE_RESET);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
editor/editor_node.cpp:1795:36: error: 'NOTIFICATION_EDITOR_POST_RESET' was not declared in this scope
 1795 |     p_node->propagate_notification(NOTIFICATION_EDITOR_POST_RESET);
------------------------------------------------------------------------------
I cant propagate any notification here, including NOTIFICATION_EDITOR_PRE_SAVE.

--- a/scene/gui/control.cpp
+++ b/scene/gui/control.cpp
@@ -1383,6 +1383,15 @@ void Control::_set_position(const Point2 &p_point) {
 
 void Control::set_position(const Point2 &p_point, bool p_keep_offsets) {
 	ERR_MAIN_THREAD_GUARD;
+
+#ifdef TOOLS_ENABLED
+	// Can't compute anchors, set position directly and return immediately.
+	if (saving && !is_inside_tree()) {
+		data.pos_cache = p_point;
+		return;
+	}
+#endif
+
 	if (p_keep_offsets) {
 		_compute_anchors(Rect2(p_point, data.size_cache), data.offset, data.anchor);
 	} else {
@@ -1440,6 +1449,14 @@ void Control::set_size(const Size2 &p_size, bool p_keep_offsets) {
 	if (new_size.y < min.y) {
 		new_size.y = min.y;
 	}
+
+#ifdef TOOLS_ENABLED
+	// Can't compute anchors, set size directly and return immediately.
+	if (saving && !is_inside_tree()) {
+		data.size_cache = new_size;
+		return;
+	}
+#endif
 
 	if (p_keep_offsets) {
 		_compute_anchors(Rect2(data.pos_cache, new_size), data.offset, data.anchor);
@@ -3140,6 +3157,14 @@ Control *Control::make_custom_tooltip(const String &p_text) const {
 void Control::_notification(int p_notification) {
 	ERR_MAIN_THREAD_GUARD;
 	switch (p_notification) {
+#ifdef TOOLS_ENABLED
+		case NOTIFICATION_EDITOR_PRE_SAVE: {
+			saving = true;
+		} break;
+		case NOTIFICATION_EDITOR_POST_SAVE: {
+			saving = false;
+		} break;
+#endif
 		case NOTIFICATION_POSTINITIALIZE: {
 			data.initialized = true;
 

--- a/scene/gui/control.h
+++ b/scene/gui/control.h
@@ -47,6 +47,10 @@ class ThemeContext;
 class Control : public CanvasItem {
 	GDCLASS(Control, CanvasItem);
 
+#ifdef TOOLS_ENABLED
+	bool saving = false;
+#endif
+
 public:
 	enum Anchor {
 		ANCHOR_BEGIN = 0,

